### ISSUE TITLE
post wallet secret input hang

### ISSUE THREAD

**Issue by therippening** [xrp-community/codius-install#1]
After entering wallet secret the script hangs.  I let it sit for 10 minutes. When I hit enter a 2nd time it shows this output.  
```
codius-install.sh: line 102: read: `[?] Running moneyd in ripple testnet ? [Y/n] ': not a valid identifier
[root@codius ~]# 
```
I have tried entering "Y"  before hitting enter but same issues.

---------------------
 OS INFO
---------------------
[root@codius ~]# uname -a
Linux codius.zerpgames.com 3.10.0-862.el7.x86_64 #1 SMP Fri Apr 20 16:44:24 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux

[root@codius ~]# cat /etc/centos-release
CentOS Linux release 7.5.1804 (Core) 
[root@codius ~]#

**Comment by therippening** [comment]
Interesting enough the following output moves past that line.  
```

[+] What is your XRP wallet secret (need for moneyd) ?
Wallet Secret: d
[+] What is your Email address ?
Email: 

```

**Comment by N3TC4T** [comment]
Hey 

can you test it again ?

**Comment by therippening** [comment]
I sure can.  However the commits haven't changed from 18 to 19.  last commit was 7 hours ago.

**Comment by therippening** [comment]
Im wondering if its me....

```

[root@codius zerpadmin]# wget https://raw.githubusercontent.com/xrp-community/codius-install/master/codius-install.sh -O codius-install.sh && bash codius-install.sh
--2018-06-10 08:13:45--  https://raw.githubusercontent.com/xrp-community/codius-install/master/codius-install.sh
Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 151.101.0.133, 151.101.64.133, 151.101.128.133, ...
Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|151.101.0.133|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 8093 (7.9K) [text/plain]
Saving to: â€˜codius-install.shâ€™

100%[======================================================================================================================================================>] 8,093       --.-K/s   in 0s      

2018-06-10 08:13:45 (56.8 MB/s) - â€˜codius-install.shâ€™ saved [8093/8093]
Welcome to codius installer!

I need to ask you a few questions before starting the setup.
You can leave the default options and just press enter if you are ok with them.

[+] First, provide the IPv4 address of the network interface
IP address: 2.2.2.2

This server is behind NAT. What is the public IPv4 address?
Public IP address: 1.1.1.1
[+] What is your Codius hostname?
Hostname: codius.zerpgames.com
[+] What is your XRP wallet secret (need for moneyd) ?
Wallet Secret: XXXXXXXXXXXXXXXXXXXXXXXXX
[+] What is your Email address ?
Email: xxxxxx@xxxx.com
[root@codius zerpadmin]# 
```

**Comment by N3TC4T** [comment]
Intresting , I think there is a problem with github , maybe it's cache the files , can you check the file after downloading and see if it's same as the current file in github ?

**Comment by therippening** [comment]
I just tested it from another site using a brand new install.  Same result as above.   Below is lengthly but I hope it helps. 

I checked the hyperd section and it is commit 19.  For some reason a new install of cent os 7 and yum update are not allowing this whole program to run after filling out the questionnaire.  Fails here just before actually trying to install dependency packages >> **sudo yum install -y gcc-c++ make epel-release git wget**

_**OUTPUT FOR REFERENCE - bullets should be + but github translated them**_
`Email: contact@zerpgames.com
+ [[ -z contact@zerpgames.com ]]
+ coloredEcho '\n[!] Installing required packages ...\n' green
+ local 'exp=\n[!] Installing required packages ...\n'
+ local color=green
+ [[ green =~ \^\[0-9]\$ ]]
+ case $(echo $color | tr '[:upper:]' '[:lower:]') in
++ echo green
++ tr '[:upper:]' '[:lower:]'
+ color=2
+ tput setaf 2
[root@app05 ~]# `

```
**SNIPPET OF CODE SECTION WHERE IT FAILS.**
# Hyperd ==============================================
coloredEcho "\n[!] Installing required packages ...\n" green
sudo yum install -y gcc-c++ make epel-release git wget
coloredEcho "\n[!] Installing Hyperd ...\n" green
curl -sSl https://coiltest.s3.amazonaws.com/upload/latest/hyper-bootstrap.sh | bash
```
**EXTRA INFO - MANUAL APPLICATION**
Some notes on another issue when trying to pull manually the hyper-bootstrap.sh file.   I am getting the following:   `curl: (23) Failed writing body (3653 != 10835)`
Perhaps adding tac to the end as such?
`curl -sSl https://coiltest.s3.amazonaws.com/upload/latest/hyper-bootstrap.sh | tac | tac | bash`

Even after pulling it down I get no response from bash.  It seems to run then fail immediately with no warning.  
`[root@app05 ~]# curl -sSl https://coiltest.s3.amazonaws.com/upload/latest/hyper-bootstrap.sh | tac | tac | bash`
`[root@app05 ~]# `

**RUNNING IN DEBUG BASH -x"** 

`[root@app05 ~]# **curl -sSl https://coiltest.s3.amazonaws.com/upload/latest/hyper-bootstrap.sh | tac | tac | bash -x**
+++ dirname bash
++ cd .
++ pwd
+ BASE_DIR=/root
+ cd /root
+ SLEEP_SEC=10
+ set -e
++ id -un
+ CURRENT_USER=root
+ BOOTSTRAP_DIR=/tmp/hyper-bootstrap-root
+ BASH_C='bash -c'
+ S3_URL=https://s3-us-west-1.amazonaws.com/coiltest/upload/latest/
+ PKG_FILE=hyper-latest.tgz
+ UNTAR_DIR=hyper-pkg
+ SUPPORT_EMAIL=support@hyper.sh
+ CENTOS7_QEMU_HYPER=qemu-hyper-2.4.1-3.el7.centos.x86_64
+ CENTOS7_HYPERSTART=hyperstart-1.0.0-1.el7.centos.x86_64
+ CENTOS7_HYPER=hyper-container-1.0.0-1.el7.centos.x86_64
+ FC23_HYPERSTART=hyperstart-1.0.0-1.fc25.x86_64
+ FC23_HYPER=hyper-container-1.0.0-1.fc25.x86_64
+ DEBIAN_HYPER=hypercontainer_0.8.1-1_amd64
+ DEBIAN_HYPERSTART=hyperstart_0.8.1-1_amd64
+ SUPPORT_DISTRO=(debian ubuntu fedora centos linuxmint)
+ LINUX_MINT_CODE=(rafaela rebecca qiana)
+ UBUNTU_CODE=(trusty utopic vivid wily xenial)
+ DEBIAN_CODE=(jessie wheezy)
+ CENTOS_VER=(6 7)
+ FEDORA_VER=(20 21 22 23 24 25)
++ tput setaf 1
+ RED=
[root@app05 ~]# `

Below is a snippet from hyper-bootstrap.sh before it hits the main().  Notice in the debug above that it fails at "RED=" however above that is the value of variable RED= tput setaf 1.  Im not sure if this is bash misprinting it or if bash is reading the script wrong or if thats how it is displayed normally.

**SNIPPET FROM THE HYPER-BOOTSTRAP.SH CODE.**
`########## Constant ##########
SUPPORT_DISTRO=(debian ubuntu fedora centos linuxmint)
LINUX_MINT_CODE=(rafaela rebecca qiana)
UBUNTU_CODE=(trusty utopic vivid wily xenial)
DEBIAN_CODE=(jessie wheezy)
CENTOS_VER=(6 7)
FEDORA_VER=(20 21 22 23 24 25)
#Color Constant
RED=`tput setaf 1`
GREEN=`tput setaf 2`
YELLOW=`tput setaf 3`
BLUE=`tput setaf 4`
WHITE=`tput setaf 7`
LIGHT=`tput bold `
**RESET=`tput sgr0`**
#Error Message`

**Comment by therippening** [comment]
We can close this out.  The issue was related to the hyper-bootstrap.sh file in this instance.  The variables RED, BLUE, etc were using "old syntax" and my CentOS 7 did not like the ' ' but instead wanted ()

**Comment by NodeStatic** [comment]
@therippening Hey man, I see that you closed this, but I am running into the issue. Not very experienced with code or debugging at all, but I am very interested in learning more about the syntax as well as dabbling with a codius host. Is there any way you could point me in the right direction for solving this issue? It would be much appreciated!
